jax/_src/scipy/linalg.py

```diff
@@ -1041,3 +1041,43 @@ def toeplitz(c: ArrayLike, r: Optional[ArrayLike] = None) -> Array:
       (nrows,), (1,), 'VALID', dimension_numbers=('NTC', 'IOT', 'NTC'),
       precision=lax.Precision.HIGHEST)[0]
   return jnp.flip(patches, axis=0)
+
+
+_NULLSPACE_DOC = """\
+Because the size of the output of ``null_space`` is data-dependent, the function is not
+typically compatible with JIT. The JAX version adds the optional ``size`` argument which
+must be specified statically for ``jax.scipy.linalg.null_space`` to be used within some of JAX's
+transformations.
+"""
+_NULLSPACE_EXTRA_PARAMS = """
+size : int, optional
+    If specified, ``size`` elements of the null space are returned. If there are
+    fewer null space elements than ``size`` indicates, the return value will be padded with constant
+    ``fill_value`` vectors.
+fill_value : array_like, optional
+    When ``size`` is specified and there are fewer null space elements than the indicated number of elements,
+    the return value will be padded with constant ``fill_value`` vectors. If ``fill_value`` is not specified,
+    it defaults to zero (indeed constant 0 vectors are always in the null space).
```

Comment:
I wonder if we need `fill_value` here? I think a more natural fill choice is to use the already computed eigenvectors. Maybe simpler to not add `fill_value` at all?

Reply:
In that case, should the fill value default to zero and not be user-specifiable? In what situations would a fill value other than zero be relevant?

Reply:
yes I think so will change to a value of 0 and clarify in the docs